game.py

```diff
@@ -0,0 +1,145 @@
+        font_size = 50
+        self.font1 = pygame.font.Font(font_path, font_size)
+
+        self.score = 0
+
+        self.clock = pygame.time.Clock()
+        self.last_spawned = pygame.time.get_ticks()
+
+        self.sounds = {
+            "jump": pygame.mixer.Sound("data/sound/jump.wav"),
+            "score": pygame.mixer.Sound("data/sound/score.wav"),
+            "collide": pygame.mixer.Sound("data/sound/fall.wav")
+        }
+
+    def get_pane_coordinates(self, rows):
+        coordinates = []
+        pane_height = self.bg_rect.height / len(rows)
+        for i in range(len(rows)):
+            for j in range(rows[i]):
+                pane_width = self.bg_rect.width / rows[i]
+                origin = [self.offset[0] + pane_width * j, pane_height * i]
+                if j > 0 and j <= rows[i] - 1:
+                    origin[0] += self.border_width / 2
+                end = origin.copy()
+                end[0] += pane_width
+                end[1] += pane_height
+                if (j <= rows[i] - 1):
+                    end[0] -= self.border_width / 2
+                coordinates.append([origin, end])
+        return coordinates  # Array of [top-left coords, bottom-right coords]
+
+    def show_score(self):
+        scores = self.font1.render(str(self.score), 1, (0, 0, 0))
+        self.gameDisplay.blit(scores, (200 + 650, 30))
+
+    def play_sound(self, sound_name):
+        self.sounds[sound_name].play()
+
+    def run(self):
+        while self.running:
+            # Gtk events
+            while Gtk.events_pending():
+                Gtk.main_iteration()
+
+            for event in pygame.event.get():
+                if event.type == pygame.QUIT:
+                    return
+                # Manage jumps
+                for i, key in enumerate(self.keymap):
+                    if event.type == pygame.KEYDOWN and event.key == key:
+                        self.play_sound("jump")
+                        self.guys[i].move()
+
+            # Assign speed as per score
+            self.speed = 7 + self.score // 8
+            self.spike_spawn_delay = ((self.display_rect.height / 2)
+                                      / self.speed) * 18
+
+            self.gameDisplay.fill(black)
+            self.gameDisplay.blit(self.background, self.offset)
+
+            # Generate spikes and increase score
+            if self.last_spawned + self.spike_spawn_delay < pygame.time.get_ticks():
+                self.generator.generate(self.speed)
+                self.score += 1
+                self.play_sound("score")
+                self.last_spawned = pygame.time.get_ticks()
+
+            self.show_score()
+
+            # updates
+            if self.generator.update(self.guys):
+                # returns true if player is dead
+                return self.score
+            for guy in self.guys:
+                guy.update()
+
+            pygame.display.update()
+            self.clock.tick(60)
```

Comment:
If this is a generic game class, then it shouldn't implement run as any new game mode should implement it's run function where it'll handle all pygame events as this just handles one.

If they're parts of this function that's generic then let those be left and this class can be inherited and any function redefinition can happen if necessary.

Reply:
The class is not supposed to inherit from, when I say it's generic I mean that it acts as all the game types itself depending upon the keymap input which it takes.

Keymap also helps the class to handle the event function accordingly in a modular manner thus eliminating a reason for any inhertied classes to implement their own run function or even inherit from it in the first place.

Reply:
Keymap doesn't help the class handle events accordingly as keymaps is a list and event handling from your changes are happening within the `pygame.event.get()` loop which at the time only handled one event.

Reply:
I've made a commit related to this [0b6a605](https://github.com/sugarlabs/make-them-fall-activity/pull/25/commits/0b6a605c6ecef4bc87a1582e08c5375c2fa8c4d3)

Reply:
Thanks!